File: data/multilang_comparison/program_go/src/func42.go
package main
import "math/rand"
var _ = rand.Intn
func Func42(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array18 *Array = Func43(&params0, loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array18.RefC--
      if array18.RefC == 0 {
          array18.Data = nil
          array18 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array20 *Array = Func44(&params0, loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array20.RefC--
      if array20.RefC == 0 {
          array20.Data = nil
          array20 = nil
      }
   }
   var params0 ArrayParam
   params0.Size = 0
   params0.Data = make([]*Array, params0.Size)
   var array22 *Array = Func45(&params0, loopsFactor);
   params0.Data = nil
   params0.Size = 0
   return array22;
}

File: data/multilang_comparison/program_julia/src/func42.jl
include("program_julia_head.jl")
function func42(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      params0 = JArray_param{Int}()
      params0.size = 0
      array18 = func43(params0, loopsFactor);
      array18.refC -= 1
      if array18.refC == 0
          empty!(array18.data)
          array18 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array20 = func44(params0, loopsFactor);
      array20.refC -= 1
      if array20.refC == 0
          empty!(array20.data)
          array20 = nothing
      end
   end
   params0 = JArray_param{Int}()
   params0.size = 0
   array22 = func45(params0, loopsFactor);
   return array22;
end

